framework/Optimizers/stepManipulators/ConjugateGradient.py

```diff
@@ -0,0 +1,436 @@
+    """
+      Determines if the current trajectory is following another trajectory.
+      @ In, traj, int, integer identifier for trajectory that needs to be checked
+      @ In, opt, dict, DENORMALIZED most recent optimal point for trajectory
+      @ In, info, dict, additional information about optimal point
+      @ In, dataObject, DataObject.DataSet, data collected through optimization so far (SolutionExport)
+      @ In, followers, list(int), trajectories that are following traj currently
+      @ In, tolerance, float, termination distance (in scaled space)
+      @ Out, found, int, trajectory that traj is following (or None)
+    """
+    if followers is None:
+      followers = []
+    # we define a trajectory as following if its current opt point is sufficiently near other opt
+    # points from other trajectories
+    matchDict = dict((var, opt[var]) for var in self._optVars)
+    # only look in accepted points #TODO would there be value in looking at others?
+    matchDict['CG_task'] = 'CONV'
+    # only look at other trajectories that this trajectory hasn't killed
+    noMatchDict = {'trajID': [traj] + followers}
+
+    _, found = dataObject.realization(matchDict=matchDict, noMatchDict=noMatchDict, tol=tolerance)
+    if found is not None:
+      return found['trajID']
+    return None
+
+  ###################
+  # Utility Methods #
+  ###################
+
+  def _polakRibierePowellStep(self, prevGrad, curGrad, gradDotProduct, searchVector):
+    """
+      TODO rename all these variables!
+      @ In, prevGrad, np.array, previous gradient in order of sampled variables
+      @ In, curGrad, np.array, current gradient in order of sampled variables
+      @ In, gradDorProduct, float, scalar product of the current grad with itself
+      @ In, searchVector, np.array, ongoing search vector (not unit vector)
+      @ Out, gNorm, float, norm of the gradient
+    """
+    deltaGradient = curGrad - prevGrad
+    gain = max(0, np.dot(deltaGradient, curGrad) / gradDotProduct)
+    searchVector = -curGrad + gain * searchVector
+    searchVectorMag = mathUtils.calculateMultivectorMagnitude(searchVector)
+    return searchVectorMag, searchVector
+
+  def modifyAcceptance(self, oldPoint, oldVal, newPoint, newVal):
+    """
+      Allows modification of acceptance criteria.
+      Note this is only called if self.needsAccessToAcceptance is True.
+      @ In, oldPoint, dict, old opt point
+      @ In, oldVal, float, old objective function value
+      @ In, newPoint, dict, new opt point
+      @ In, newVal, float, new objective function value
+      @ In, info, dict, identifying information about evaluation
+      @ Out, accept, boolean, whether we store the point
+    """
+    # Because in ConjugateGradient we use all the line search information,
+    ## we "accept" all points from the Optimizer's standpoint, and allow
+    ## the step manipulator to use the information.
+    return 'accepted'
```

Comment:
accept is not boolean...it is a string?

Reply:
yes, the acceptance status is descriptive, not boolean.